paper/src/main/java/net/draycia/carbon/paper/integration/fuuid/TruceChannel.java

```diff
@@ -0,0 +1,127 @@
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.checkerframework.framework.qual.DefaultQualifier;
+import org.spongepowered.configurate.objectmapping.ConfigSerializable;
+
+import static net.draycia.carbon.api.channels.ChannelPermissionResult.channelPermissionResult;
+
+@DefaultQualifier(NonNull.class)
+@ConfigSerializable
+public class TruceChannel extends AbstractFactionsChannel {
+
+    public static final String FILE_NAME = "factionsuuid-trucechat.conf";
+
+    private transient @MonotonicNonNull @Inject CarbonMessages messages;
+    private transient @MonotonicNonNull @Inject UserManager<?> users;
+
+    public TruceChannel() {
+        this.key = Key.key("carbon", "trucechat");
+        this.commandAliases = List.of("tc");
+
+        this.messageSource = new ConfigChannelMessageSource();
+        this.messageSource.defaults = Map.of(
+            "default_format", "(truce: %factionsuuid_faction_name%) <display_name>: <message>",
+            "console", "[truce: %factionsuuid_faction_name%] <username>: <message>"
+        );
+    }
+
+    @Override
+    public ChannelPermissionResult speechPermitted(final CarbonPlayer player) {
+        return channelPermissionResult(
+            this.hasRelations(player, Relation.TRUCE),
+            () -> this.messages.cannotUseTruceChannel(player)
+        );
+    }
+
+    @Override
+    public ChannelPermissionResult hearingPermitted(final CarbonPlayer player) {
+        return channelPermissionResult(
+            this.hasRelations(player, Relation.TRUCE),
+            () -> this.messages.cannotUseTruceChannel(player)
+        );
+    }
+
+    @Override
+    public List<Audience> recipients(final CarbonPlayer sender) {
+        if (!this.hasRelations(sender, Relation.TRUCE)) {
+            if (sender.online()) {
+                sender.sendMessage(this.messages.cannotUseTruceChannel(sender));
+            }
+
+            return Collections.emptyList();
+        }
+
+        final List<Audience> recipients = new ArrayList<>();
+        for (final Player player : this.hasTruceWith(sender)) {
+            final @Nullable CarbonPlayer carbon = this.users.user(player.getUniqueId()).getNow(null);
+            if (carbon != null) {
+                recipients.add(carbon);
+            }
+        }
+
+        recipients.add(this.server.console());
+
+        return recipients;
+    }
+
+    private List<Player> hasTruceWith(final CarbonPlayer player) {
+        final @Nullable Faction faction = this.faction(player);
+
+        if (faction == null) {
+            return List.of();
+        }
+
+        final List<Player> alliedPlayers = new ArrayList<>();
+
+        for (final FPlayer onlinePlayer : FPlayers.getInstance().getOnlinePlayers()) {
+            final Relation relation = faction.getRelationTo(onlinePlayer);
+
+            // Not sure if checking MEMBER is required, or if we only need to check ALLY
+            if (relation == Relation.MEMBER || relation == Relation.TRUCE) {
```

Comment:
same here `relation.atLeast(Relation.TRUCE)` this will include MEMBER & ALLY & TRUCE